functorch/csrc/BatchRulesDecompositions.cpp

```diff
@@ -131,6 +210,8 @@ TORCH_LIBRARY_IMPL(aten, FT_BATCHED_KEY, m) {
   OP_DECOMPOSE2(unflatten, int);
   OP_DECOMPOSE(cross_entropy_loss);
   OP_DECOMPOSE(arctan2);
+  DECOMPOSE_FUNCTIONAL(diag_embed);
+  DECOMPOSE_FUNCTIONAL(block_diag);
```

Comment:
Does `vmap(vmap(diag_embed))(x)` work?

Reply:
I just tried it locally. It looks like it works:
```
>>> import torch
>>> from functorch import vmap
>>> print(vmap(vmap(torch.diag_embed))(torch.ones(2, 2, 2)))
/scratch/hirsheybar/pytorch-functionalization-env/lib/python3.6/site-packages/functorch-0.0.1a0+20d7754-py3.6-linux-x86_64.egg/functorch/_src/vmap.py:319: UserWarning: There is a performance drop because we have not yet implemented the batching rule for aten::diagonal_scatter. Please file us an issue on GitHub so that we can prioritize its implementation. (Triggered internally at  /raid/hirsheybar/pytorch-functionalization/functorch/functorch/csrc/BatchedFallback.cpp:79.)
  batched_outputs = func(*batched_inputs, **kwargs)
tensor([[[[1., 0.],
          [0., 1.]],

         [[1., 0.],
          [0., 1.]]],


        [[[1., 0.],
          [0., 1.]],

         [[1., 0.],
          [0., 1.]]]])
>>>
```